### PR TITLE
[DebugInfo] Don't emit .loc directive with all values zero

### PR DESCRIPTION
**Pull request by ostannard**
When emitting debug info for code alignment, it was possible to emit a .loc directive with a file number of zero, which is invalid for DWARF 4 and earlier. This happened because getCurrentDwarfLoc() returned a zero-initialised value when there hadn't been a previous .loc directive emitted.

### CODE REVIEW

File: llvm/test/DebugInfo/ARM/align-func-start.ll [hunk 1]
```diff
@@ -0,0 +1,26 @@
+; RUN: llc -mtriple=arm-none-eabi < %s | FileCheck %s
+; RUN: llc -mtriple=arm-none-eabi < %s | llvm-mc --triple=arm-none-eabi -mcpu=cortex-m3
+
+; CHECK-NOT: .loc    0 0 0
+; CHECK: .loc    1 2 3 prologue_end
```
Comment (pogo59) [inline llvm/test/DebugInfo/ARM/align-func-start.ll:5]:
```suggestion
; CHECK-NOT: .loc    0
; CHECK: .loc    1 2 3 prologue_end
; CHECK-NOT: .loc 0
```
This is slightly more robust, and means you don't need the second RUN line.

**Review by jthackray — APPROVED** [review]
LGTM

File: llvm/test/DebugInfo/ARM/align-func-start.ll [hunk 2]
```diff
@@ -0,0 +1,27 @@
+; RUN: llc -mtriple=arm-none-eabi < %s | FileCheck %s
+; RUN: llc -mtriple=arm-none-eabi < %s | llvm-mc --triple=arm-none-eabi -mcpu=cortex-m3
```
Comment (OCHyams) [inline llvm/test/DebugInfo/ARM/align-func-start.ll:2]: Do we need this 2nd run line?

**Review by OCHyams — APPROVED** [review]
Seems like a reasonable fix, LGTM, thanks

File: llvm/test/DebugInfo/ARM/align-func-start.ll (hunk 2 above)
Comment (ostannard) [reply llvm/test/DebugInfo/ARM/align-func-start.ll:2]: It checks for the original way we discovered this problem, which was that `llc` generated assembly which was rejected by the assembly parser.

Comment (OCHyams) [reply llvm/test/DebugInfo/ARM/align-func-start.ll:2]: Ah I see, ok cool. Potentially worth adding a comment saying as such? ymmv, still LGTM either way

Comment (ostannard) [reply llvm/test/DebugInfo/ARM/align-func-start.ll:2]: Done, thanks.

**Pull request merged**